Engine/ResourceManagement/Resources/StateMachine.cpp

```diff
@@ -0,0 +1,252 @@
+#include "StateMachine.h"
+#include "Main/Application.h"
+#include "Module/ModuleFileSystem.h"
+#include "Module/ModuleResourceManager.h"
+#include "Helper/Config.h"
+
+//Structs
+
+Clip::Clip(std::string& name, std::shared_ptr<Animation>& animation, bool loop) :
+	name(name), name_hash(std::hash<std::string>{}(name)), animation(animation), loop(loop), animation_time((animation->frames / animation->frames_per_second) * 1000) {}
+void Clip::SetAnimation(const std::shared_ptr<Animation>& animation)
+{
+	this->animation = animation;
+	animation_time = (animation->frames / animation->frames_per_second) * 1000;
+}
+
+
+State::State(std::string name, std::shared_ptr<Clip> clip) :
+	name(name), name_hash(std::hash<std::string>{}(name)),clip(clip)  {
+}
```

Comment:
name should be passed by reference

Reply:
if I want to use it like State("blabla", nullprt); like I'm doing I need it to be like that without reference